### ISSUE TITLE
Error when removing expiration date

### ISSUE THREAD

**Issue by aronovgj** [nextcloud/server#1119]
### Steps to reproduce
1. Set automatic expiration date for public links to 1 week in settings
2. Share a link
3. Unset expiration date for the file you shared
### Expected behaviour

No error logs.
### Actual behaviour

Error occurs

```
Error   no app in context   $params["date"] was missing. Transferred value: Array ( [itemType] => file [itemSource] => 6268 [date] => [uidOwner] => me) 
```

Link and everything else still works though.
### Server configuration

Mint 17.3x64

**Web server:**
Nginx 1.11.1
**Database:**
mysql 5.5.50
**PHP version:**
PHP 7.0.10
**Nextcloud version:** (see Nextcloud admin page)
10.0
**Updated from an older Nextcloud/ownCloud or fresh install:**
Update from Nextcloud 9.0.53
**Where did you install Nextcloud from:**
https://download.nextcloud.com/server/releases/nextcloud-10.0.0.zip

**Signing status:**
<details>
<summary>

Signing status</summary>

```
No errors have been found.
```

</details>

**List of activated apps:**
<details>
<summary>

App list</summary>

```
Enabled:
  - activity: 2.3.2
  - admin_audit: 1.0.0
  - calendar: 1.3.3
  - comments: 1.0.0
  - contacts: 1.3.1.0
  - dav: 1.0.0
  - direct_menu: 0.9.0
  - documents: 0.13.1
  - federatedfilesharing: 1.0.1
  - federation: 1.0.1
  - files: 1.5.2
  - files_pdfviewer: 0.8.1
  - files_retention: 1.0.0
  - files_sharing: 1.0.0
  - files_texteditor: 2.1
  - files_trashbin: 1.0.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 15.0.0
  - logreader: 1.1.5
  - notifications: 0.3.0
  - password_policy: 1.0.0
  - provisioning_api: 1.0.0
  - serverinfo: 1.1.1
  - survey_client: 0.1.5
  - systemtags: 1.0.2
  - templateeditor: 0.1
  - theming: 1.0.1
  - updatenotification: 1.0.1
  - workflowengine: 1.0.1
```

</details>

**The content of config/config.php:**
<details>
<summary>

Config report</summary>

```
{
    "system": {
        "instanceid": "ochgoz86g7jo",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "xxx.xxx.xxx.xxx",
            "www.me.org",
            "me.org",
            "localhost"
        ],
        "datadirectory": "\/home\/me\/nextcloud\/data",
        "overwrite.cli.url": "https:\/\/me.org",
        "dbtype": "mysql",
        "version": "9.1.0.16",
        "dbname": "nextcloud",
        "dbhost": "localhost",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379,
            "password": "***REMOVED SENSITIVE VALUE***",
            "timeout": 0,
            "dbindex": 0
        },
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "Europe\/Berlin",
        "loglevel": 2,
        "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
        "log_rotate_size": 10485760,
        "installed": true,
        "asset-pipeline.enabled": false,
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_from_address": "me",
        "mail_domain": "me.net",
        "mail_smtpport": "465",
        "mail_smtphost": "mail.me.net",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "theme": ""
    }
}
```

</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no
### Client configuration

**Browser:**
Firefox
**Operating system:**
Windows 7

**Comment by rullzer** [comment]
I can't reproduce this. 

What does your browser console tell you is send?

**Comment by aronovgj** [comment]
```
Headers
Params
POST
Response
Cookies
Call Stack
Cache-Controlno-store, no-cache, must-revalidateConnectionkeep-aliveContent-Security-Policydefault-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *Content-Typeapplication/json; charset=utf-8DateSun, 04 Sep 2016 12:56:26 GMTExpiresThu, 19 Nov 1981 08:52:00 GMTPragmano-cacheServernginxStrict-Transport-Securitymax-age=15768000;    # includeSubDomains; preload;Transfer-EncodingchunkedX-Content-Type-OptionsnosniffX-Download-OptionsnoopenX-Frame-OptionsSAMEORIGINX-Permitted-Cross-Domain-PoliciesnoneX-Robots-TagnoneX-XSS-Protection1; mode=block
Acceptapplication/json, text/javascript, */*; q=0.01Accept-Encodinggzip, deflate, brAccept-Languageen-US,en;q=0.5Connectionkeep-aliveContent-Length123Content-Typeapplication/x-www-form-urlencoded; charset=UTF-8Cookienc_sameSiteCookielax=true; nc_sameSiteCookiestrict=true; ochgoz86g7jo=hpt9ordq9a5j21pu8o23tk9im7; oc_sessionPassphrase=OgN8Iib0w5yVEVKi7%2FXb3xBi173%2FcqkU6FyTTzr%2BPoUi0NFt6vHn7OHMyfs5N19NE22RkRmC39JG1%2Bou1tOeRnkOyrw2ad1P6mlHMay9rCw7FSdQYfHDRc4LidPLlr1vDNT1Hostmyaddress.orgOCS-APIREQUESTtrueUser-AgentMozilla/5.0 (Windows NT 6.1; Win64; x64; rv:51.0) Gecko/20100101 Firefox/51.0X-Requested-WithXMLHttpRequestrequesttokenHxggGgJBJD1YA2QJKE8bBBxDAy4rE21XFCMNI3MmTTQ=:GpaBEpOm9JR8FwiTH0QhJDZfDRIPGL4RTJAKtkfYmmg=

GET 
XHR 
https://myaddress.org/ocs/v2.php/apps/files_sharing/api/v1/shares [HTTP/1.1 200 OK 140ms]
Headers
Params
Response
Cookies
Call Stack
Cache-Controlno-store, no-cache, must-revalidateConnectionkeep-aliveContent-Security-Policydefault-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *Content-Typeapplication/json; charset=utf-8DateSun, 04 Sep 2016 12:56:26 GMTExpiresThu, 19 Nov 1981 08:52:00 GMTPragmano-cacheServernginxStrict-Transport-Securitymax-age=15768000;    # includeSubDomains; preload;Transfer-EncodingchunkedX-Content-Type-OptionsnosniffX-Download-OptionsnoopenX-Frame-OptionsSAMEORIGINX-Permitted-Cross-Domain-PoliciesnoneX-Robots-TagnoneX-XSS-Protection1; mode=block
Accept*/*Accept-Encodinggzip, deflate, brAccept-Languageen-US,en;q=0.5Connectionkeep-aliveCookienc_sameSiteCookielax=true; nc_sameSiteCookiestrict=true; ochgoz86g7jo=hpt9ordq9a5j21pu8o23tk9im7; oc_sessionPassphrase=OgN8Iib0w5yVEVKi7%2FXb3xBi173%2FcqkU6FyTTzr%2BPoUi0NFt6vHn7OHMyfs5N19NE22RkRmC39JG1%2Bou1tOeRnkOyrw2ad1P6mlHMay9rCw7FSdQYfHDRc4LidPLlr1vDNT1Hostmyaddress.orgOCS-APIREQUESTtrueUser-AgentMozilla/5.0 (Windows NT 6.1; Win64; x64; rv:51.0) Gecko/20100101 Firefox/51.0X-Requested-WithXMLHttpRequestrequesttokenHxggGgJBJD1YA2QJKE8bBBxDAy4rE21XFCMNI3MmTTQ=:GpaBEpOm9JR8FwiTH0QhJDZfDRIPGL4RTJAKtkfYmmg=

PUT 
XHR 
https://myaddress.org/ocs/v2.php/apps/files_sharing/api/v1/shares/32 [HTTP/1.1 200 OK 139ms]
Headers
Params
POST
Response
Cookies
Call Stack
Cache-Controlno-store, no-cache, must-revalidateConnectionkeep-aliveContent-Security-Policydefault-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *Content-Typeapplication/json; charset=utf-8DateSun, 04 Sep 2016 12:56:30 GMTExpiresThu, 19 Nov 1981 08:52:00 GMTPragmano-cacheServernginxStrict-Transport-Securitymax-age=15768000;    # includeSubDomains; preload;Transfer-EncodingchunkedX-Content-Type-OptionsnosniffX-Download-OptionsnoopenX-Frame-OptionsSAMEORIGINX-Permitted-Cross-Domain-PoliciesnoneX-Robots-TagnoneX-XSS-Protection1; mode=block
Acceptapplication/json, text/javascript, */*; q=0.01Accept-Encodinggzip, deflate, brAccept-Languageen-US,en;q=0.5Connectionkeep-aliveContent-Length11Content-Typeapplication/x-www-form-urlencoded; charset=UTF-8Cookienc_sameSiteCookielax=true; nc_sameSiteCookiestrict=true; ochgoz86g7jo=hpt9ordq9a5j21pu8o23tk9im7; oc_sessionPassphrase=OgN8Iib0w5yVEVKi7%2FXb3xBi173%2FcqkU6FyTTzr%2BPoUi0NFt6vHn7OHMyfs5N19NE22RkRmC39JG1%2Bou1tOeRnkOyrw2ad1P6mlHMay9rCw7FSdQYfHDRc4LidPLlr1vDNT1Hostmyaddress.orgOCS-APIREQUESTtrueUser-AgentMozilla/5.0 (Windows NT 6.1; Win64; x64; rv:51.0) Gecko/20100101 Firefox/51.0X-Requested-WithXMLHttpRequestrequesttokenHxggGgJBJD1YA2QJKE8bBBxDAy4rE21XFCMNI3MmTTQ=:GpaBEpOm9JR8FwiTH0QhJDZfDRIPGL4RTJAKtkfYmmg=

Strict-Transport-Security: The site specified a header that could not be parsed successfully.[Learn More]  32
GET 
XHR 
https://myaddress.org/ocs/v2.php/apps/files_sharing/api/v1/shares [HTTP/1.1 200 OK 141ms]
Headers
Params
Response
Cookies
Call Stack
Cache-Controlno-store, no-cache, must-revalidateConnectionkeep-aliveContent-Security-Policydefault-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *Content-Typeapplication/json; charset=utf-8DateSun, 04 Sep 2016 12:56:30 GMTExpiresThu, 19 Nov 1981 08:52:00 GMTPragmano-cacheServernginxStrict-Transport-Securitymax-age=15768000;    # includeSubDomains; preload;Transfer-EncodingchunkedX-Content-Type-OptionsnosniffX-Download-OptionsnoopenX-Frame-OptionsSAMEORIGINX-Permitted-Cross-Domain-PoliciesnoneX-Robots-TagnoneX-XSS-Protection1; mode=block
Accept*/*Accept-Encodinggzip, deflate, brAccept-Languageen-US,en;q=0.5Connectionkeep-aliveCookienc_sameSiteCookielax=true; nc_sameSiteCookiestrict=true; ochgoz86g7jo=hpt9ordq9a5j21pu8o23tk9im7; oc_sessionPassphrase=OgN8Iib0w5yVEVKi7%2FXb3xBi173%2FcqkU6FyTTzr%2BPoUi0NFt6vHn7OHMyfs5N19NE22RkRmC39JG1%2Bou1tOeRnkOyrw2ad1P6mlHMay9rCw7FSdQYfHDRc4LidPLlr1vDNT1Hostmyaddress.orgOCS-APIREQUESTtrueUser-AgentMozilla/5.0 (Windows NT 6.1; Win64; x64; rv:51.0) Gecko/20100101 Firefox/51.0X-Requested-WithXMLHttpRequestrequesttokenHxggGgJBJD1YA2QJKE8bBBxDAy4rE21XFCMNI3MmTTQ=:GpaBEpOm9JR8FwiTH0QhJDZfDRIPGL4RTJAKtkfYmmg=

GET 
XHR 
https://myaddress.org/ocs/v2.php/apps/notifications/api/v1/notifications [HTTP/1.1 200 OK 130ms]
Headers
Params
Response
Cookies
Call Stack
Cache-Controlno-store, no-cache, must-revalidateConnectionkeep-aliveContent-Security-Policydefault-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *Content-Typeapplication/json; charset=utf-8DateSun, 04 Sep 2016 12:56:52 GMTEtagd751713988987e9331980363e24189ceExpiresThu, 19 Nov 1981 08:52:00 GMTPragmano-cacheServernginxStrict-Transport-Securitymax-age=15768000;    # includeSubDomains; preload;Transfer-EncodingchunkedX-Content-Type-OptionsnosniffX-Download-OptionsnoopenX-Frame-OptionsSAMEORIGINX-Permitted-Cross-Domain-PoliciesnoneX-Robots-TagnoneX-XSS-Protection1; mode=block
Accept*/*Accept-Encodinggzip, deflate, brAccept-Languageen-US,en;q=0.5Connectionkeep-aliveCookienc_sameSiteCookielax=true; nc_sameSiteCookiestrict=true; ochgoz86g7jo=hpt9ordq9a5j21pu8o23tk9im7; oc_sessionPassphrase=OgN8Iib0w5yVEVKi7%2FXb3xBi173%2FcqkU6FyTTzr%2BPoUi0NFt6vHn7OHMyfs5N19NE22RkRmC39JG1%2Bou1tOeRnkOyrw2ad1P6mlHMay9rCw7FSdQYfHDRc4LidPLlr1vDNT1Hostmyaddress.orgOCS-APIREQUESTtrueUser-AgentMozilla/5.0 (Windows NT 6.1; Win64; x64; rv:51.0) Gecko/20100101 Firefox/51.0X-Requested-WithXMLHttpRequestrequesttokenHxggGgJBJD1YA2QJKE8bBBxDAy4rE21XFCMNI3MmTTQ=:GpaBEpOm9JR8FwiTH0QhJDZfDRIPGL4RTJAKtkfYmmg=
```

**Comment by MorrisJobke** [comment]
This was afaik fixed with 10.0.1 - cc @schiessle
